monitoring/definitions/postgres.go

```diff
@@ -0,0 +1,119 @@
+							Warning:           monitoring.Alert().LessOrEqual(5).For(5 * time.Minute),
+							PossibleSolutions: "none",
+						},
+						monitoring.Observable{
+							Name:              "transactions",
+							Description:       "transaction_durations",
+							Owner:             monitoring.ObservableOwnerCloud,
+							Query:             "sum by (datname) (pg_stat_activity_max_tx_duration{datname!~\"template.*|postgres|cloudsqladmin\"})",
+							DataMayNotExist:   true,
+							DataMayNotBeNaN:   false,
+							Warning:           monitoring.Alert().GreaterOrEqual(300).For(5 * time.Minute),
+							Critical:          monitoring.Alert().GreaterOrEqual(500).For(5 * time.Minute),
+							PanelOptions:      monitoring.PanelOptions().Unit(monitoring.Milliseconds),
+							PossibleSolutions: "none",
+						},
+					},
+				},
+			},
+			{
+				Title: "Database and collector status", Hidden: true,
+				Rows: []monitoring.Row{{
+					monitoring.Observable{
+						Name:              "postgres_up",
+						Description:       "current db status",
+						Owner:             monitoring.ObservableOwnerCloud,
+						Query:             "pg_up",
+						DataMayNotExist:   true,
+						DataMayNotBeNaN:   true,
+						Critical:          monitoring.Alert().LessOrEqual(0).For(5 * time.Minute),
+						PossibleSolutions: "none",
+					},
+					monitoring.Observable{
+						Name:              "pg_exporter_err",
+						Description:       "error scraping postgres exporter",
+						Owner:             monitoring.ObservableOwnerCloud,
+						Query:             "pg_exporter_last_scrape_error",
+						DataMayNotExist:   true,
+						DataMayNotBeNaN:   true,
+						Warning:           monitoring.Alert().GreaterOrEqual(1).For(5 * time.Minute),
+						PossibleSolutions: "none",
+					},
+					monitoring.Observable{
+						Name:            "database_migration_status",
+						Description:     "schema migration status",
+						Owner:           monitoring.ObservableOwnerCloud,
+						Query:           "pg_sg_db_migrations",
+						DataMayNotExist: true,
+						Critical:        monitoring.Alert().GreaterOrEqual(1).For(5 * time.Minute),
+						PossibleSolutions: "A database migration has been in progress for 5 minutes," +
+							" ensure that the migration has succeeded or contact Sourcegraph",
+					},
+					monitoring.Observable{
+						Name:              "cache_hit_ratio",
+						Description:       "ratio of cache hits (should be 99%)",
+						Owner:             monitoring.ObservableOwnerCloud,
+						Query:             "avg(rate(pg_stat_database_blks_hit{datname!~\"template.*|postgres|cloudsqladmin\"}[5m]) / (rate(pg_stat_database_blks_hit{datname!~\"template.*|postgres|cloudsqladmin\"}[5m]) + rate(pg_stat_database_blks_read{datname!~\"template.*|postgres|cloudsqladmin\"}[5m]))) by (datname)",
+						DataMayNotExist:   false,
+						DataMayNotBeNaN:   false,
+						Warning:           monitoring.Alert().LessOrEqual(0.98).For(5 * time.Minute),
+						PossibleSolutions: "none",
```

Comment:
```suggestion
						PossibleSolutions: "Cache hit ratio should be 99%.",
```

(I'm not sure what someone would want to look at should this not be the case, might want to add it here)

Reply:
I think this is mostly for internal use. We should strive to keep this high otherwise we should add new indexes.

Reply:
Regardless, we should add a note here since we ship this to customers. I think adding new indexes is a valid possible solution, though since itl need action on our end maybe we could recommend admins open an issue?

Reply:
That's a great idea!